src/dbus/dbus_thread_object.cpp

```diff
@@ -0,0 +1,449 @@
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CHANNEL_PROPERTY,
+                               std::bind(&DBusThreadObject::GetChannelHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_MASTER_KEY_PROPERTY,
+                               std::bind(&DBusThreadObject::GetMasterKeyHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CCA_FAILURE_RATE_PROPERTY,
+                               std::bind(&DBusThreadObject::GetCcaFailureRateHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LINK_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetLinkCountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_IP6_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetIp6CountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_SUPPORTED_CHANNEL_MASK_PROPERTY,
+                               std::bind(&DBusThreadObject::GetSupportedChannelMaskHandler, this, _1));
+
+    return err;
+}
+
+void DBusThreadObject::DeviceRoleHandler(otDeviceRole aDeviceRole)
+{
+    SignalPropertyChanged(OTBR_DBUS_PREFIX, OTBR_DBUS_DEVICE_ROLE_PROPERTY, GetDeviceRoleName(aDeviceRole));
+}
+
+void DBusThreadObject::ScanHandler(DBusRequest &aRequest)
+{
+    auto &threadHelper = mNcp->GetThreadHelper();
+    if (threadHelper->Scan([aRequest](const std::vector<otActiveScanResult> &aResult) mutable {
+            aRequest.Reply(std::tie(aResult));
+        }) != OT_ERROR_NONE)
+    {
+        std::vector<otActiveScanResult> result {};
+
+        aRequest.Reply(std::tie(result));
+    }
+}
+
+void DBusThreadObject::ReplyScanResult(DBusRequest &aRequest, const std::vector<otActiveScanResult> &aResult)
+{
+    aRequest.Reply(std::tie(aResult));
+}
+
+void DBusThreadObject::AttachHandler(DBusRequest &aRequest)
+{
+    auto &               threadHelper = mNcp->GetThreadHelper();
+    std::string          name;
+    uint16_t             panid;
+    uint64_t             extPanId;
+    std::vector<uint8_t> masterKey;
+    std::vector<uint8_t> pskc;
+    uint32_t             channelMask;
+
+    auto args = std::tie(masterKey, panid, name, extPanId, pskc, channelMask);
+
+    if (DBusMessageToTuple(*aRequest.GetMessage(), args) != OTBR_ERROR_NONE)
+    {
+        aRequest.ReplyOtResult(OT_ERROR_INVALID_ARGS);
+    }
+    else
+    {
+        threadHelper->Attach(name, panid, extPanId, masterKey, pskc, channelMask,
+                             [aRequest](otError aError) mutable { aRequest.ReplyOtResult(aError); });
+    }
+}
+
+void DBusThreadObject::FactoryResetHandler(DBusRequest &aRequest)
+{
+    auto &threadHelper = mNcp->GetThreadHelper();
+
+    mNcp->Reset();
+    aRequest.ReplyOtResult(OT_ERROR_NONE);
+    threadHelper->AddDeviceRoleHandler(std::bind(&DBusThreadObject::DeviceRoleHandler, this, _1));
```

Comment:
use lambda? as we do in function `AttachHandler`.

Reply:
The function is a little big here so I considered to use bind, or it's preferred to call a member function inside the lambda?

Reply:
No such preference, just want to keep consistent with surrounding functions (such as AttachHandler).